tests/test_vision_tagger_telemetry.py

```
@@ -0,0 +1,171 @@
+        payload = _parse_log(caplog)
+        assert payload["error"] == "fetch_rejected"
+
+    def test_gemini_quota(self, caplog):
+        caplog.set_level(logging.WARNING, logger=LOGGER_NAME)
+        with patch.object(
+            vision_tagger, "_fetch_image", return_value=(b"x", "image/jpeg")
+        ), patch.object(vision_tagger, "_client") as mock_client:
+            mock_client.return_value.models.generate_content.side_effect = Exception(
+                "RESOURCE_EXHAUSTED: quota exceeded"
+            )
+            with pytest.raises(Exception):
```

Ruff (0.15.12)

[warning] Do not assert blind exception: `Exception`

(B017)